### PR TITLE
Fix/house keeping and bug fixes

### PR DESCRIPTION
**Pull request by craigyu**
# Description
This PR is not associated with any particular Jira issue, the content of this PR is to hand off my current progress before I shift my focus to making our mono repo. 

### Summary of Changes:
- implement the registration steps validation logic skeleton 
- modularize ModalStateManager previously in `frontend/src/views/Seedlot/SeedlotRegistrationForm/index.tsx`
- modularize the submit popup previously in `frontend/src/views/Seedlot/SeedlotRegistrationForm/index.tsx`
- remove eslint ignore comment in test files
- force `mockServerHost` to be `REACT_APP_NRSPARWEBAPP_VERSION` in prod @RMCampos I think this will fix the prod crashing for sure now 👀
    - image
    - I think it's due to recent-activities endpoint returning an HTML file without mock server running
- Ownership validation state is now a prop from the parent API, so it will not crash, this should fix [FSADT2-618](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-618)
- fix Ownership portion calculation, tested with 3 owners each having: 33.33, 33.34, 33.33. This should fix [FSADT2-615](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-615)
  

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?
Test locally and manually

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-118-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-118-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by craigyu** [comment]
WOAH [27 Code Smells](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=118&resolved=false&types=CODE_SMELL) 🤧🤧🤧, @DerekRoberts probably turned on some sonarcloud features. We should discuss if we can turn some off, for example with the current eslint and sonarcloud setting we cannot use for loops at all, and I'm not a fan of using `forEach` everywhere.

- Eslint doesn't allow me to use for ... of 
    - Screenshot 2023-05-02 at 9 38 12 AM
- Sonarcloud doesn't let me use simple for loops
    - image

**Comment by craigyu** [comment]
@RMCampos @lucasbuenoencora @mgaseta @ArthurEncr 
It's ready for review :)

File: frontend/src/api-service/ApiConfig.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ const serverHost = env.REACT_APP_SERVER_URL || mockServerConfig.namespace;
 
 const oracleServerHost = env.REACT_APP_ORACLE_SERVER_URL || mockServerConfig.namespace;
 
-const mockServerHost = mockServerConfig.namespace;
+const isProdEnv = env.REACT_APP_NRSPARWEBAPP_VERSION?.startsWith('prod');
```
Comment (RMCampos) [inline frontend/src/api-service/ApiConfig.ts:8]: 👍🏻

File: frontend/src/components/SeedlotRegistrationSteps/OwnershipStep/utils.ts [hunk 2]
```diff
@@ -215,7 +205,7 @@ export const arePortionsValid = (ownershiptArray: Array<SingleOwnerForm>): boole
   ownershiptArray.forEach((obj) => {
     sum += Number(obj.ownerPortion);
   });
-  return sum === 100;
+  return Number(sum.toFixed(2)) === 100;
```
Comment (RMCampos) [inline frontend/src/components/SeedlotRegistrationSteps/OwnershipStep/utils.ts:208]: 👍🏻

**Review by RMCampos — APPROVED** [review]
Great job! 🥇 🥳 LGTM

**Pull request merged**